### PR TITLE
Update build status reporting details

### PR DESCRIPTION
**Pull request by BenWhitehead**
Update build status reporting details to include the new statuses that
will be reported starting in 1.11 and provide a short description of
what each status represents.

### CODE REVIEW

**Comment by BenWhitehead** [comment]
@mesosphere-mergebot bump-ee

File: README.md [hunk 1]
```diff
@@ -227,16 +227,44 @@ storage:
...
-Pull requests automatically trigger a new build and several tests.
+Pull requests automatically trigger a new build and several runs of the tests.
+
+In order for a pull request to be eligible to merge all required status checks must be in a successful state.
+
+#### `teamcity/dcos/build/dcos`
+Builds `dcos_generate_config.sh` and accompanying AWS CloudFormation and Azure ARM templates.  Triggered builds ran in [TeamCity](https://teamcity.mesosphere.io/project.html?projectId=DcOs_Open_Build_BuildDcOs&tab=projectOverview), run [build_teamcity](https://github.com/dcos/dcos/blob/master/build_teamcity). To run the same operations in a local development environment use [build_local.sh](https://github.com/dcos/dcos/blob/master/build_local.sh) (see above).
+
+#### `teamcity/dcos/build/tox`
+Run `tox` in the top-level dir to run all syntax checks as well as pytest (unit-tests). See [tox.ini](https://github.com/dcos/dcos/blob/master/tox.ini) for more details
```
Comment (orsenthil) [inline README.md]: Runs

File: README.md [hunk 2]
```diff
@@ -227,16 +227,44 @@ storage:
...
+
+#### `teamcity/dcos/test/aws/cloudformation/simple`
+Deploy a DC/OS Cluster using the AWS Simple CloudFormation template generated in `teamcity/dcos/build/dcos` and run tests in [`dcos-integration-test`](packages/dcos-integration-test).
```
Comment (orsenthil) [inline README.md]: Deploys

File: README.md [hunk 3]
```diff
@@ -227,16 +227,44 @@ storage:
...
+
+#### `teamcity/dcos/test/aws/cloudformation/zen`
+Deploy a DC/OS Cluster using the AWS Zen CloudFormation template generated in `teamcity/dcos/build/dcos` and run tests in [`dcos-integration-test`](packages/dcos-integration-test).
```
Comment (orsenthil) [inline README.md]: Deploys

File: README.md [hunk 4]
```diff
@@ -227,16 +227,44 @@ storage:
...
+
+#### `teamcity/dcos/test/aws/onprem/static`
+Use [dcos-launch](https://github.com/dcos/dcos-launch) to create an AWS VPC Cluster of CentOS 7 servers and then run the installer bundled the `dcos_generate_config.sh` generated in `teamcity/dcos/build/dcos` to install DC/OS. Following successful installation run tests in [`dcos-integration-test`](packages/dcos-integration-test).
```
Comment (orsenthil) [inline README.md]: Action verbs should be `Uses`, `Deploys`, `Runs` etc in this document.

The information provided looks good to me.

**Comment by jgehrcke** [comment]
Great stuff! @orsenthil can  you give this another look? After you are happy I will also quickly review and then let's push this forward. Thanks!

**Comment by jgehrcke** [comment]
Ship-it was applied prematurely. See DCOS-19252.

**Pull request closed without merging**